src/Dfe.PlanTech.AzureFunctions/Mappings/RecommendationChunkMapper.cs

```diff
@@ -34,17 +31,15 @@ public override async Task PostUpdateEntityCallback(MappedEntity mappedEntity)
 
         if (existing != null)
         {
-            existing.Answers.AddRange(await _db.RecommendationChunkAnswers
-                                                .Where(recChunkAnswer => recChunkAnswer.AnswerId == incoming.Id)
-                                                .Select(recChunkAnswer => recChunkAnswer.Answer)
-                                                .Select(content => content!)
-                                                .ToListAsync());
-
-            existing.Content.AddRange(await _db.RecommendationChunkContents
-                                                .Where(recChunkContent => recChunkContent.RecommendationChunkId == incoming.Id)
-                                                .Select(recChunkContent => recChunkContent.ContentComponent)
-                                                .Select(content => content!)
-                                                .ToListAsync());
+            if (existing.Answers == null || existing.Answers.Count == 0)
+            {
+                await _db.RecommendationChunkAnswers.IgnoreQueryFilters().Where(recChunkAnswer => recChunkAnswer.RecommendationChunkId == existing.Id).Include(rca => rca.Answer).ToListAsync();
```

Comment:
I might be having a moment.. but whats this doing? its not assigned to anything so looks like a void statement?

Reply:
It does indeed look like a void statement, and it is indeed not assigned to anything.

_However_, behind the scenes:

1. EF Core is tracking the recommendation chunk (since it was retrieved from the database)
2. EF Core knows that the `Answers` field on the `RecommendationChunkDbEntity` uses the join table `RecommendationChunkAnswers`
3. As a result, when `RecommendationChunkAnswers` are retrieved, it will _automatically_ add them to the `Answers` field of the appropriate recommendation chunk. I.e. in this instance, the above command will result in `existing.Answers` being automatically populated with the result of that `await _db...` command.

I realise it's _implict_ rather than _explict_ and, to be honest, originally I had something like

`existing.Answers = await _db.RecommendationChunkAnswers.IgnoreQueryFilters().Where(recChunkAnswer => recChunkAnswer.RecommendationChunkId == existing.Id).Select(rca => rca.Answer).ToListAsync();`

But then EF Core _doesn't know that those Answers are already attached to the recommendation chunk entity` because we've missed out the join table.

So the other three solutions I considered are:
1. Retrieve the `Answers` on initial query, which requires a far bit more code changes unfortunately due to how the rest of the code is currently organised
2. Do something like

```csharp
var recommendationChunkAnswers= await _db.RecommendationChunkAnswers.IgnoreQueryFilters().Where(recChunkAnswer => recChunkAnswer.RecommendationChunkId == existing.Id).Include(rca => rca.Answer).ToListAsync();

existingEntity.Answers = recommendationChunkAnswers.Select(recChunkAnswer => recChunkAnswer.Answer).ToList()
```

3. Add the join table to the `RecommendationChunkDbEntity` as well, but this again requires more code changes than ideal.

As a result I went with the above solution as it is the simplest, albeit requires some behind the scenes knowledge of how EF Core works.

Reply:
Ah thanks for running through all this. Option 2 clarifies this more I think but I see the point of it being longer, I think this would benefit from comments to explain the tracking, in case a future dev doesn't know this

Reply:
Sorted 👍